DPNUnet.imjoy.html

```diff
@@ -513,7 +516,81 @@
+        Data has to be stored on the plugin engine where DPNUnet plugin is running.
+        Segmented nuclei will also be converted in GeoJson regions, and (if requested)
+        a download link with the zipped results be provided. 
+
+        Data has to be stored in the following format:
+        - Each image has to be in its own folder (sample folder)
+        - Sample folders must be stored in parental folder called 'test'
+        - Each image has to have the same name (e.g. 'nuclei.png')
+
+        Args:
+            ctx.data.data_path (string): folder containing data to be segmented (folder containing subfolder 'test').
+            ctx.data.image_file_name (string): name of images to be segmented.
+            ctx.data.zip_name (string): results can be provided as a zip archive. Specify name here, no zip is created when None. 
+
+        Returns:
+            results_url (string): url pointing to the zipped results of the segmentation
+        '''
+
+        # Get arguments
+        data_path =  ctx.data.data_path
+        image_file_name =  ctx.data.image_file_name
+        zip_name =  ctx.data.zip_name
+
+        # Check if model is imported
+        model_path = os.path.join('..', 'weights', 'dpn_softmax_f0', 'fold0_best.pth') # '../weights/dpn_softmax_f0/fold0_best.pth'
+        if not os.path.exists(model_path):
+            api.alert('No model found. Please import model before launching prediction.')
+            return
+
+        # Check if data path is present
+        if not os.path.exists(data_path):
+            api.alert('Data path does not exist. ')
+            return 
+
+        # Configuration for prediction    
+        config = load_config('configs/imjoy_dpn_softmax_s2.json')
+        config.image_file_name = image_file_name
+        config.dataset_path = os.path.join(data_path,'test')
+        config.results_dir  = os.path.join(data_path, '__outputs__')
+        config.batch_size = 16
+        config.epoch_size = 24
+        scale_factor = 1
+  
+        # Perform prediction
+        paths, fn_mapping  = load_path_mapping(config, False)
+        self.dash = await api.createWindow(type="Im2Im-Dashboard", name="DPN-Unet Segmentation Model", w=25, h=10, data={'display_mode': 'all'})
+        try:
+            await loop.run_in_executor(None, eval_bowl, self.dash, config, paths, fn_mapping, model_path, self.plot_images,scale_factor)
+        except Exception as e:
+            api.alert(f'Error occured: {e}')
+
+        # Create geojson 
+        for root, dirnames, filenames in os.walk(config.dataset_path):
+            for filename in filenames:
+                if 'border_mask_output_labels' in filename: 
+                    img_label = imread(os.path.join(root,filename))
+                    segmentationUtils.masks_to_polygon(img_label, label='nuclei', simplify_tol=0,
+                                               save_name=os.path.join(root, "annotation.json"))   
+
+        # Create ZIP of results
```

Comment:
For this part, since it's not really close related to zip, I suggest to wrap as another api function, perhaps called `export_result()`.  Maybe it's not a good idea to zip the entire dataset folder, because it may contain the original input file and also other temporary files. So you may want to allow filtering the files inside the files, only indluce those result files, this can also be done by passing an config dictionary to configure what do you want to be exported into the zip file.

Then the return value of `predict` can be the result folder path.